concoction-lib/src/main/java/info/mmpa/concoction/Concoction.java

```diff
@@ -1,3 +1,87 @@
 package info.mmpa.concoction;
 
-public class Concoction {}
+import info.mmpa.concoction.model.ModelSource;
+import info.mmpa.concoction.model.impl.BasicApplicationModel;
+import info.mmpa.concoction.output.Results;
+import info.mmpa.concoction.scan.model.insn.InstructionsMatchingModel;
+import info.mmpa.concoction.scan.standard.StandardScan;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * The main API entrypoint for concoction.
+ * This is a builder to get a {@link Results} object.
+ * Currently, this only implements static scanning as dynamic scanning is not hardened yet.
+ */
+public class Concoction {
+    private final List<Path> modelPaths = new ArrayList<>();
+
+    private ModelSource primarySource;
+
+    private final List<ModelSource> supportingSources = new ArrayList<>();
+
+    private Concoction() {}
+
+    /**
+     * The initialization point of the builder
+     * @return A Concoction builder
+     */
+
+    public static Concoction builder() {
+        return new Concoction();
+    }
+
+    /**
+     * Adds a model path to the current builder
+     * @param path The path of the JSON model to add.
+     * @return The updated builder
```

Comment:
Nit: `Self` is shorter and more accurate, the instance is not changing. Applies to other self-returns.